src/main/java/uk/gov/hmcts/reform/hmc/helper/hmi/CaseClassificationsMapper.java

```diff
@@ -1,37 +1,108 @@
+        !StringUtils.isEmpty(caseCategory.getCategoryParent());
 
     public List<CaseClassification> getCaseClassifications(CaseDetails caseDetails) {
-        List<CaseClassification> caseClassifications = new ArrayList<>();
-
-        val caseClassificationType = caseDetails.getCaseCategories().stream().filter(
-            element -> element.getCategoryType().equalsIgnoreCase(CaseCategoryType.CASETYPE.getLabel())
-        ).findFirst().orElseThrow(() -> new BadRequestException(CASE_TYPE_ERROR));
-
-        val caseClassificationSubType = caseDetails.getCaseCategories().stream().filter(
-            element -> element.getCategoryType().equalsIgnoreCase(CaseCategoryType.CASESUBTYPE.getLabel())
-        ).findFirst().orElseThrow(() -> new BadRequestException(CASE_SUBTYPE_ERROR));
-
-        val caseClassificationBuilder = CaseClassification.builder()
-            .caseClassificationService(caseDetails.getHmctsServiceCode());
-        caseClassificationBuilder.caseClassificationType(caseClassificationType.getCategoryValue());
-        caseClassificationBuilder.caseClassificationSubType(caseClassificationSubType.getCategoryValue());
-        caseClassifications.add(caseClassificationBuilder.build());
-        return caseClassifications;
+
+        val caseClassificationTypes = getCaseCategories(caseDetails, CaseCategoryType.CASETYPE);
+        validateSize(caseClassificationTypes,  new BadRequestException(CASE_TYPE_ERROR));
+        validateCategoriesOfCaseType(caseClassificationTypes);
+
+        val caseTypeValues = caseClassificationTypes.stream()
+            .map(element -> element.getCategoryValue())
+            .collect(Collectors.toList());
+
+        val caseClassificationSubTypes = getCaseCategories(caseDetails, CaseCategoryType.CASESUBTYPE);
+        validateSize(caseClassificationSubTypes,  new BadRequestException(CASE_SUBTYPE_ERROR));
+        validateCaseTypeAndCaseSubType(caseTypeValues, caseClassificationSubTypes);
+
+        return caseClassificationSubTypes.stream()
+            .map(caseClassificationSubType ->
+                     buildCaseClassification(
+                         caseDetails.getHmctsServiceCode(),
+                         caseClassificationSubType,
+                         caseTypeValues
+                     )
+            ).collect(Collectors.toList());
+    }
+
+    private void validateCaseTypeAndCaseSubType(List<String> caseTypeValues,
+                                                List<CaseCategory> caseClassificationSubTypes) {
+
+        val categoryParents = caseClassificationSubTypes.stream()
+            .map(element -> element.getCategoryParent())
+            .collect(Collectors.toList());
+
+        caseTypeValues.stream()
+            .forEach(caseType -> {
+                if (!categoryParents.contains(caseType)) {
+                    throw new BadRequestException(CASE_TYPE_CASE_SUBTYPE_ERROR);
+                }
+            });
+        return;
```

Comment:
This is a void, we don't need the return statement

Reply:
Agreed